### PR TITLE
[API 7] Setting Tooltip

### LINKED ISSUE #56: [API 7] Setting tooltip
Add annotation param `tooltip` to `Setting`

```java
    public @interface Tooltip {
        String title() default "";
        String body();
        String docUrl() default "";
    }
    Tooltip tooltip() default @Tooltip(body = "");
```

### PR DESCRIPTION
**Pull request by Pjiesco**
Closes #56

### CODE REVIEW

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java [hunk 1]
```diff
@@ -34,6 +34,17 @@ public static Pair<JsonObject, TypeSpec.Builder> process(TouchPortalPluginAnnota
         jsonSetting.addProperty(SettingHelper.TYPE, desiredTPType);
         jsonSetting.addProperty(SettingHelper.DEFAULT, setting.defaultValue());
         jsonSetting.addProperty(SettingHelper.IS_READ_ONLY, setting.isReadOnly());
+
+        if (!setting.tooltip().body().isEmpty()) {
+            JsonObject tooltip = new JsonObject();
+
+            tooltip.addProperty(SettingHelper.Tooltip.TITLE, setting.tooltip().title());
```
Comment (ChristopheCVB) [inline AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java:41]: Check if title is not empty before adding it?

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java [hunk 2]
```diff
@@ -34,6 +34,17 @@ public static Pair<JsonObject, TypeSpec.Builder> process(TouchPortalPluginAnnota
...
+            tooltip.addProperty(SettingHelper.Tooltip.BODY, setting.tooltip().body());
+            tooltip.addProperty(SettingHelper.Tooltip.DOC_URL, setting.tooltip().docUrl());
```
Comment (ChristopheCVB) [inline AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java:43]: Check if docUrl is not empty before adding it?

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java [hunk 3]
```diff
@@ -66,6 +77,8 @@ public static Pair<JsonObject, TypeSpec.Builder> process(TouchPortalPluginAnnota
                 throw new GenericHelper.TPTypeException.Builder(className).typeUnsupported(desiredTPType).forAnnotation(GenericHelper.TPTypeException.ForAnnotation.SETTING).build();
         }
 
+
```
Comment (ChristopheCVB) [inline AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java:80]: Remove those empty lines

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java (hunk 1 above)
Comment (Pjiesco) [reply AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java:41]: Should we do the same here? https://github.com/ChristopheCVB/TouchPortalPluginSDK/blob/a527952202346bbedc3d3ecafb1592b34da377cd/AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/utils/SpecUtils.java#L215

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java (hunk 2 above)
Comment (Pjiesco) [reply AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java:43]: Should we do the same here? https://github.com/ChristopheCVB/TouchPortalPluginSDK/blob/a527952202346bbedc3d3ecafb1592b34da377cd/AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/utils/SpecUtils.java#L217

File: AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java (hunk 1 above)
Comment (ChristopheCVB) [reply AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/SettingProcessor.java]: > Should we do the same here?
> 
> https://github.com/ChristopheCVB/TouchPortalPluginSDK/blob/a527952202346bbedc3d3ecafb1592b34da377cd/AnnotationsProcessor/src/main/java/com/christophecvb/touchportal/annotations/processor/utils/SpecUtils.java#L215

Would make sense indeed

**Review by ChristopheCVB — APPROVED** [review]

**Pull request merged**